.github/workflows/deploy-optimized.yml

```
@@ -0,0 +1,557 @@
+            fi
+          else
+            echo "ℹ️ No migration secret configured"
+            echo "status=skipped" >> $GITHUB_OUTPUT
+          fi
+
+  # Rollback on failure
+  rollback-on-failure:
+    name: 🔄 Auto Rollback
+    runs-on: ubuntu-latest
+    needs: [deploy, validate-deployment]
+    if: failure() && needs.deploy.outputs.deployment_id != '' && env.TARGET_ENV == 'production'
```

actionlint (1.7.7)

context "env" is not allowed here. available contexts are "github", "inputs", "needs", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)